hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractJsonGetter.java

```diff
@@ -104,44 +104,62 @@ Object getValue(Object obj) {
     }
 
     @Override
-    Object getValue(Object obj, String attributePath) {
-        JsonPathCursor pathCursor = getPath(attributePath);
+    public Object getValue(Object obj, String attributePath) {
+        try (JsonParser parser = createParser(obj)) {
+            return getValue(obj, attributePath, parser);
+        } catch (IOException e) {
+            // Just return null in case of exception. Json strings are allowed to be invalid.
+            return null;
+        }
+    }
 
+    public Object[] getValues(Object obj, String[] attributePaths) {
         try (JsonParser parser = createParser(obj)) {
-            parser.nextToken();
-            while (pathCursor.getNext() != null) {
-                if (pathCursor.isArray()) {
+            final Object[] values = new Object[attributePaths.length];
```

Comment:
the parser is not restarted after each attribute. will it work for multiple attributes?